pkg/storage/sharding/label.go

```diff
@@ -100,3 +105,36 @@ func ParseShardIDLabelValue(val string) (index, shardCount uint64, _ error) {
 
 	return index - 1, count, nil
 }
+
+// ShardFunc is a simpler API to the above, when you don't need to avoid allocations.
+func ShardFunc(l labels.Labels) uint64 {
+	hash, _ := labelsXXHash(make([]byte, 0, 1024), l)
+	return hash
+}
+
+// labelsXXHash replicates historical hash computation from Prometheus circa 2022.
+// The buffer b can be recycled to avoid allocations.
+func labelsXXHash(b []byte, ls labels.Labels) (uint64, []byte) {
```

Comment:
Why do we need this? If it's still the same, why don't we just call `ls.Hash()`?

Reply:
Bryan is going to change the `Labels.Hash()` function in Prometheus, if the labels-as-string implementation will end up there.

Reply:
I think that we can rely on that function now with a test that checks that it didn't change, and bring this implementation here once it changes.